action.yaml

```diff
@@ -0,0 +1,70 @@
+name: 'Setup IOG devx'
+description: 'Adds a `devx` shell includes `ghc` and `cabal`'
+inputs:
+  platform:
+    description: 'One of x86_64-linux, x86_64-darwin, aarch64-linux, aarch64-darwin'
+    default: 'x86_64-linux'
+
+  target-platform:
+    description: 'One of "" (native), "-static", "-windows" or "-js"'
+    default: ''
+
+  compiler-nix-name:
+    description: 'The version of GHC to use'
+    default: 'ghc961'
+
+  minimal:
+    description: 'Set to `false` to get hlint and hls'
+    default: true
+
+  iog:
+    description: 'Set to `true` to get libsodum, libsecp256k1 and libblst'
+    default: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install Nix with good defaults
+      uses: cachix/install-nix-action@v17
+      with:
+        extra_nix_config: |
+          trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ= loony-tools:pr9m4BkM/5/eSTZlkQyRt57Jz7OMBxNSUiMC4FkcNfk= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
+          substituters = https://cache.iog.io/ https://cache.zw3rk.com/ https://cache.nixos.org/
+        nix_path: nixpkgs=channel:nixos-unstable
+    - name: Install jq
+      shell: bash
+      run: |
+        curl -L https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 -o jq
+        chmod +x jq
+    - name: Download and import closure
+      shell: bash
+      run: |
+        TOKEN=$(curl --silent https://ghcr.io/token\?scope\=repository:input-output-hk/devx:pull | jq -r .token)
+
+        BLOB=$(curl \
+          --silent \
+          --request 'GET' \
+          --header "Authorization: Bearer $TOKEN" \
+          --header "Accept: application/vnd.docker.distribution.manifest.list.v2+json" \
+          --header "Accept: application/vnd.docker.distribution.manifest.v2+json" \
+          --header "Accept: application/vnd.oci.image.manifest.v1+json" \
+          "https://ghcr.io/v2/input-output-hk/devx/manifests/${{ inputs.platform }}.${{ inputs.compiler-nix-name }}${{ inputs.target-platform }}${{ inputs.minimal && '-minimal' || '' }}${{ inputs.iog && '-iog' || '' }}-env" | tee manifest.json | jq -r '.layers[0].digest')
+
+        curl \
```

Comment:
# Download the docker image layer that contains our nix store closure, and import it.